pyttman/core/internals.py

```diff
@@ -32,6 +35,12 @@ def depr_graceful(message: str, version: str):
     out = f"{message} - This was deprecated in version {version}."
     warnings.warn(out, DeprecationWarning)
 
+class CustomUserDict(UserDict):
+     
+    # constructor
+    def __init__(self, dictionary):
+        self.data = dictionary
+        self.__dict__.update(dictionary)
```

Comment:
I think this class is redundant, since we already have the `Settings` class which implements the same API. I'd prefer if the `Settings` class would handle a recursive assignment of self for all sub-dictionaries, instead of the first level being a `Settings` object and all subnodes being of a different type, for no apparent reason.